packages/docs/docs/providers/battlenet.md

```diff
@@ -15,21 +15,100 @@ https://develop.battle.net/access/clients
 
 The **Battle.net Provider** comes with a set of default options:
 
-- [Battle.net Provider options](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/providers/battlenet.js)
+- [Battle.net Provider options](https://github.com/nextauthjs/next-auth/blob/main/src/providers/battlenet.ts)
 
 You can override any of the options to suit your own use case.
 
+### Issuer
+
+Battle.net oprrate with account partitions, an issuer should be selected according to the region: https://develop.battle.net/documentation/guides/regionality-and-apis
+
+| Region | Issuer                             |
+| ------ | ---------------------------------- |
+| US     | https://us.battle.net/oauth        |
+| EU     | https://eu.battle.net/oauth        |
+| KR     | https://kr.battle.net/oauth        |
+| TW     | https://tw.battle.net/oauth        |
+| CN     | https://www.battlenet.com.cn/oauth |
+
 ## Example
 
+### Minimal configuration
+
+```js
+import BattleNetProvider from "next-auth/providers/battlenet";
+...
+providers: [
+  BattleNetProvider({
+    clientId: process.env.BATTLENET_CLIENT_ID,
+    clientSecret: process.env.BATTLENET_CLIENT_SECRET,
+    issuer: process.env.BATTLENET_ISSUER
+  })
+]
+...
+```
+
+### Request access to WoW profile
+
 ```js
 import BattleNetProvider from "next-auth/providers/battlenet";
 ...
 providers: [
   BattleNetProvider({
     clientId: process.env.BATTLENET_CLIENT_ID,
     clientSecret: process.env.BATTLENET_CLIENT_SECRET,
-    region: process.env.BATTLENET_REGION
+    issuer: process.env.BATTLENET_ISSUER,
+    authorization: {
+      params: {
+        scope: 'openid wow.profile'
+      }
+    }
   })
 ]
 ...
 ```
+
+:::note
+`openid` should be always included in the scope list
+:::
+
+### NextAuth configuration to save the access token to the user session
+
+:::tip
+The `accessToken` can be used to fetch profile data from a specific Game
+:::
+
+```js
+...
+export default NextAuth({
+  providers: [
+    BattleNetProvider({
+      clientId: process.env.BATTLENET_CLIENT_ID,
+      clientSecret: process.env.BATTLENET_CLIENT_SECRET,
+      issuer: process.env.BATTLENET_ISSUER,
+      authorization: {
+        params: {
+          scope: 'openid wow.profile'
+        }
+      }
+    })
+  ],
+  secret: process.env.NEXTAUTH_SECRET,
+  session: {
+    strategy: 'jwt',
+    maxAge: 1 * 24 * 60 * 60 // 1 day
+  },
+  callbacks: {
+    async jwt ({ token, account }) {
+      if (account) {
+        token.accessToken = account.access_token
+      }
+      return token
+    },
+    async session ({ session, token }) {
+      session.accessToken = token.accessToken
+      return session
+    }
+  }
+})
+```
```

Comment:
This whole section is a duplicate of what's documented already in https://next-auth.js.org/configuration/callbacks and also https://next-auth.js.org/getting-started/example#using-nextauthjs-callbacks

We should just link to those sections to keep provider docs consistent.

What you could mention is the need for the extra `wow.profile` scope I assume, and link to some Battle.net documentation about it.